src/index.js

```diff
@@ -168,19 +169,18 @@ const loadPlayer = (params, resolve, reject) => {
     return;
   }
 
-  const src = getUrl(params);
-
-  // If we've already downloaded this script, we should have the proper `bc`
-  // global and can bypass the script creation process.
-  if (scriptCache.has(src)) {
+  // If we've already downloaded this script or detected a matching global, we
+  // should have the proper `bc` global and can bypass the script creation
+  // process.
+  if (cache.has(params)) {
     resolve(initPlayer(params, embed));
     return;
   }
 
   const script = document.createElement('script');
 
   script.onload = () => {
-    scriptCache.add(src);
+    cache.store(params);
```

Comment:
If this took another argument we could just pass in the url here.

Reply:
I like the idea that the cache deals in a simple set of parameters - `{accountId, playerId, embedId}` - and its internal implementation is left up to the cache itself. Today it stores the script URL (if it can), but tomorrow it may change, but the interface wouldn't have to.